pkg/uki/common.go

```diff
@@ -153,3 +159,104 @@ func copyFile(src, dst string) error {
 
 	return destinationFile.Close()
 }
+
+// checkArtifactsignatureIsValid checks that a give efi artifact is signed properly with a signature that would allow it to
+// boot correctly in the current node if secureboot is enabled
+func checkArtifactsignatureIsValid(fs v1.FS, dir string, artifact string, logger sdkTypes.KairosLogger) error {
+	var err error
+	fullArtifact := filepath.Join(dir, artifact)
+	logger.Logger.Info().Str("what", fullArtifact).Msg("Checking artifact")
+	o, err := fs.Open(fullArtifact)
+	if errors.Is(err, os.ErrNotExist) {
+		logger.Warnf("%s does not exist", fullArtifact)
+		return fmt.Errorf("%s does not exist", fullArtifact)
+	} else if errors.Is(err, os.ErrPermission) {
+		logger.Warnf("%s permission denied. Can't read file", fullArtifact)
+		return fmt.Errorf("%s permission denied. Can't read file", fullArtifact)
+	}
+
+	ok, err := checkValidEfiHeader(o)
+	_ = o.Close()
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to read file %s: %s", fullArtifact, err))
+		return fmt.Errorf("failed to read file %s: %s", fullArtifact, err)
+	}
+	if !ok {
+		logger.Error("invalid pe header")
+		return fmt.Errorf("invalid pe header")
+	}
+
+	// First check if its on dbx, if it matches, then it means that its not valid and we can return early
+	dbx, err := efi.Getdbx()
+	if err != nil {
+		logger.Logger.Error().Err(err).Msg("getting DBx certs")
+		return err
+	}
+
+	x509CertDbx, err := util.ReadCert(dbx.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// We need to read the current db database to have the proper certs to check against
+	db, err := efi.Getdb()
+	if err != nil {
+		logger.Logger.Error().Err(err).Msg("getting DB certs")
+		return err
+	}
+
+	x509CertDb, err := util.ReadCert(db.Bytes())
```

Comment:
same thing here (multiple certs possible) although we could pretend that it's a feature to only have the kairos cert in db (as in "more secure"). We can't tell the same for dbx though because one may even want to add old kairos digests or certs in dbx, so having multiple there is a good thing.